### PR TITLE
Remove deprecations for breaking release

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Fixes https://github.com/SciML/DiffEqFlux.jl/issues/707

### CODE REVIEW

File: src/DiffEqFlux.jl [hunk 1]
```diff
@@ -39,25 +31,18 @@ ZygoteRules.@adjoint ZygoteRules.literal_getproperty(A::Tridiagonal, ::Val{:du})
...
 include("collocation.jl")
 include("hnn.jl")
 include("multiple_shooting.jl")
 
-Flux.device(::FastLayer) = @warn "device(f::FastLayer) is a no-op: to move FastChain computations to a GPU, apply gpu(x) to the weight vector"
-Flux.gpu(::FastLayer) = @warn "device(f::FastLayer) is a no-op: to move FastChain computations to a GPU, apply gpu(x) to the weight vector"
-Flux.cpu(::FastLayer) = @warn "device(f::FastLayer) is a no-op: to move FastChain computations to a CPU, apply cpu(x) to the weight vector"
-
-export DeterministicCNF, FFJORD, NeuralODE, NeuralDSDE, NeuralSDE, NeuralCDDE, NeuralDAE, NeuralODEMM, TensorLayer, AugmentedNDELayer, SplineLayer, NeuralHamiltonianDE
+export DeterministicCNF, FFJORD, NeuralODE, NeuralDSDE, NeuralSDE, NeuralCDDE, NeuralDAE,
```
Comment (prbzrg) [inline src/DiffEqFlux.jl:42]: `DeterministicCNF` has been removed, so we don't have to export it.

**Comment by ChrisRackauckas** [comment]
@Abhishek-1Bhatt can you take this one from here? All that needs to be done is that the tests need to be updated in the same way as the tutorials are being updated (a lot of the tests and examples should be the same)

**Comment by Abhishek-1Bhatt** [comment]
Sure 👍

**Comment by prbzrg** [comment]
Issues related to `sciml_train` should be closed or moved after this PR merged.
https://github.com/search?q=sciml_train+repo%3ASciML%2FDiffEqFlux.jl+state%3Aopen+in%3Atitle&type=Issues

File: src/hnn.jl [hunk 2]
```diff
@@ -45,11 +45,6 @@ struct HamiltonianNN{M, R, P}
         end
         return new{typeof(model), typeof(re), typeof(p)}(model, re, p)
```
Comment (ChrisRackauckas) [inline src/hnn.jl:46]: This is missing Lux compatibility dispatches.

File: src/neural_de.jl [hunk 3]
```diff
@@ -11,19 +11,16 @@ derivatives of the loss backwards in time.
 
 ```julia
 NeuralODE(model,tspan,alg=nothing,args...;kwargs...)
```
Comment (ChrisRackauckas) [inline src/neural_de.jl:13]: Need to mention Lux

**Comment by ChrisRackauckas** [comment]
Did a lot of issue cleanup. Most were solved already.

**Comment by Abhishek-1Bhatt** [comment]
Still doesn't run full CI. Also there are some conflicts with master. Besides maybe we need to need to merge master into this branch, it says 10 commits behind.

File: test/cnf_test.jl [hunk 4]
```diff
@@ -1,4 +1,4 @@
-using DiffEqFlux, Distances, Distributions, DistributionsAD, Optimization,
+using Flux, Distances, Distributions, DistributionsAD, Optimization,
```
Comment (Abhishek-1Bhatt) [inline test/cnf_test.jl:1]:
```suggestion
using DiffEqFlux, Distances, Distributions, DistributionsAD, Optimization,
```

File: test/newton_neural_ode.jl [hunk 5]
```diff
@@ -23,22 +23,27 @@ nODE = NeuralODE(NN, tspan, ROCK4(), reltol=1e-4, saveat=[tspan[end]])
 
 loss_function(θ) = Flux.Losses.mse(y, nODE(x, θ)[end])
 l1 = loss_function(nODE.p)
-
-res = DiffEqFlux.sciml_train(loss_function, nODE.p, NewtonTrustRegion(), Optimization.AutoZygote(), maxiters=100, cb=cb) #ensure backwards compatibility of `cb`
+optf = Optimization.OptimizationFunction((p)->loss_function(p), Optimization.AutoZygote())
```
Comment (Abhishek-1Bhatt) [inline test/newton_neural_ode.jl:26]:
```suggestion
optf = Optimization.OptimizationFunction((x,p)->loss_function(x), Optimization.AutoZygote())

```

**Comment by Abhishek-1Bhatt** [comment]
This line errors the Newton NeuralODE Test
https://github.com/SciML/DiffEqFlux.jl/blob/c48a0e147bb2196ed277dcfa300ca4c90351683b/test/newton_neural_ode.jl#L51

Is this a compatibility issue with Lux?

```julia
ERROR: MethodError: no method matching initial_state(::Optim.KrylovTrustRegion{Float64}, ::Optim.Options{Float64, OptimizationOptimJL.var"#_cb#11"{var"#5#6", Optim.KrylovTrustRegion{Float64}, Base.Iterators.Cycle{Tuple{Optimization.NullData}}}}, ::NLSolversBase.TwiceDifferentiableHV{Float32, ComponentArrays.ComponentVector{Float32, Vector{Float32}, Tuple{ComponentArrays.Axis{(layer_1 = ViewAxis(1:10, Axis(weight = ViewAxis(1:5, ShapedAxis((5, 1), NamedTuple())), bias = ViewAxis(6:10, ShapedAxis((5, 1), NamedTuple())))), layer_2 = ViewAxis(11:16, Axis(weight = ViewAxis(1:5, ShapedAxis((1, 5), NamedTuple())), bias = ViewAxis(6:6, ShapedAxis((1, 1), NamedTuple())))))}}}, ComponentArrays.ComponentVector{Float32, Vector{Float32}, Tuple{ComponentArrays.Axis{(layer_1 = ViewAxis(1:10, Axis(weight = ViewAxis(1:5, ShapedAxis((5, 1), NamedTuple())), bias = ViewAxis(6:10, ShapedAxis((5, 1), NamedTuple())))), 
layer_2 = ViewAxis(11:16, Axis(weight = ViewAxis(1:5, ShapedAxis((1, 5), NamedTuple())), bias = ViewAxis(6:6, ShapedAxis((1, 1), NamedTuple())))))}}}, ComponentArrays.ComponentVector{Float32, Vector{Float32}, Tuple{ComponentArrays.Axis{(layer_1 = ViewAxis(1:10, Axis(weight = ViewAxis(1:5, ShapedAxis((5, 1), NamedTuple())), bias = ViewAxis(6:10, ShapedAxis((5, 1), NamedTuple())))), layer_2 = ViewAxis(11:16, Axis(weight = ViewAxis(1:5, ShapedAxis((1, 5), NamedTuple())), bias = ViewAxis(6:6, ShapedAxis((1, 1), NamedTuple())))))}}}}, ::ComponentArrays.ComponentVector{Float32, Vector{Float32}, Tuple{ComponentArrays.Axis{(layer_1 = ViewAxis(1:10, Axis(weight = ViewAxis(1:5, ShapedAxis((5, 1), NamedTuple())), bias = ViewAxis(6:10, ShapedAxis((5, 1), NamedTuple())))), layer_2 = ViewAxis(11:16, Axis(weight = ViewAxis(1:5, ShapedAxis((1, 5), NamedTuple())), bias = ViewAxis(6:6, ShapedAxis((1, 1), NamedTuple())))))}}})
Closest candidates are:
  initial_state(::AcceleratedGradientDescent, ::Any, ::Any, ::AbstractArray{T}) where T at C:\Users\user\.julia\packages\Optim\6Lpjy\src\multivariate\solvers\first_order\accelerated_gradient_descent.jl:35
  initial_state(::Optim.KrylovTrustRegion, ::Any, ::Any, ::Array{T}) where T at C:\Users\user\.julia\packages\Optim\6Lpjy\src\multivariate\solvers\second_order\krylov_trust_region.jl:39
  initial_state(::SimulatedAnnealing, ::Any, ::Any, ::AbstractArray{T}) where T at C:\Users\user\.julia\packages\Optim\6Lpjy\src\multivariate\solvers\zeroth_order\simulated_annealing.jl:62
```

**Comment by ChrisRackauckas** [comment]
See if you can isolate it to `Optim.KrylovTrustRegion()` just requiring standard `Array` types: that would be my guess.

**Comment by Abhishek-1Bhatt** [comment]
https://github.com/JuliaNLSolvers/Optim.jl/blob/d5cb5dae049dcb7bcff94d691a099b3650d5d9d8/src/multivariate/solvers/second_order/krylov_trust_region.jl#L39

**Comment by Abhishek-1Bhatt** [comment]
What would be the best way to allow `initial_x` above to be a ComponentVector?

**Comment by ChrisRackauckas** [comment]
Just use Flux.jl for that for now. The real answer is an upstream fix to Optim.jl but that shouldn't block this.

**Comment by avik-pal** [comment]
Just a heads up here. I will be deprecating quite a few functionalities (mostly these were undocumented but they ended up in user code) in v0.4.8 for removal in v0.5 (See https://github.com/avik-pal/Lux.jl/blob/ap/tests/src/deprecated.jl). Might be worthwhile avoiding using these (most notably `ActivationFunction`)

**Comment by ChrisRackauckas** [comment]
Continued in https://github.com/SciML/DiffEqFlux.jl/pull/794

**Pull request closed without merging**